src/app/interceptors/auth.interceptor.ts

```diff
@@ -0,0 +1,29 @@
+import { Injectable } from '@angular/core';
+import {
+  HttpRequest,
+  HttpHandler,
+  HttpEvent,
+  HttpInterceptor,
+} from '@angular/common/http';
+import { Observable } from 'rxjs';
+import { UserService } from '../services/user.service';
+
+@Injectable()
+export class AuthInterceptor implements HttpInterceptor {
+  constructor(private userService: UserService) {}
+
+  intercept(
+    request: HttpRequest<unknown>,
+    next: HttpHandler,
+  ): Observable<HttpEvent<unknown>> {
+    const token = this.userService.accessToken;
+    if (token) {
+      request = request.clone({
+        setHeaders: {
+          authorization: `Bearer ${token}`,
```

Comment:
there are still some components doing this manually, right? should they eventually switch to using this interceptor?